test/python/mock/test_mock_module_deprecation.py

```diff
@@ -0,0 +1,23 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test for deprecation of qiskit.test.mock module."""
+from qiskit.test import QiskitTestCase
+
+
+class MockModuleDeprecationTest(QiskitTestCase):
+    """Test for deprecation of qiskit.test.mock module."""
+
+    def test_deprecated_mock_module(self):
+        """Test that the mock module is deprecated."""
+        with self.assertWarns(DeprecationWarning):
+            import qiskit.test.mock  # pylint: disable=unused-import
```

Comment:
This is the single test failure, but if I actually run
```python
import qiskit
import qiskit.test
import qiskit.test.mock
```
locally, I get the deprecation warning, and only on the third line (where it's expected).  It's possible that this import statement is being triggered at some point during test discovery.  I did a quick search and spotted an import in `test/python/compiler/test_scheduler.py`, which is probably the culprit.

Reply:
Thanks for checking the status. Sorry I couldn't get back to this PR because I was quite busy this week. I have planned to fix the error and add documentation next Monday and Tuesday. Would that give you sufficient review time to meet the RC1 date (Thursday)?

Reply:
That's fine for me, no worries.